src/convertToObject.js

```diff
@@ -3,16 +3,36 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const solutionObject = {};
+  const sourceArray = sourceString
+    .split(';') // split the string into an array of names and their values
+    .map(el => el.split(':'))
+    // split the elements into arrays of a key and value each
+    .filter(el => el.length > 1) // remove the invalid elements
+    .map(el => [ // clear up line breaks and spaces
```

Comment:
Maybe you could change the naming of some "el" just to add some clarity
```suggestion
    .map(line => [ // clear up line breaks and spaces
```